### PR TITLE
feat: Implement dropdown component with search

### PR DESCRIPTION
**Pull request by memoyil**
@Chef-Chungus I implemented as we spoke with minor differences, instead of selecting the item when searching, I filtered out the other items which as a user perspective makes more sense. I would like to hear your opinion as well.

### CODE REVIEW

File: packages/pancake-uikit/src/components/DropdownSearch/DropdownSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
+import React, { useState, useRef, useEffect } from "react";
+import styled, { css } from "styled-components";
+import { Input } from "../Input";
+import { Text } from "../Text";
+import { ArrowDropDownIcon } from "../Svg";
+import { DropdownSearchOptionProps, DropdownSearchProps } from "./types";
+
+const StyledInput = styled(Input)`
+  border-radius: 16px;
```
Comment (hachiojidev) [inline packages/pancake-uikit/src/components/DropdownSearch/DropdownSearch.tsx:9]: Input already has a border radius of 16px

File: packages/pancake-uikit/src/components/DropdownSearch/DropdownSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+      onChange(option);
+    }
+    setContent(options);
+    if (searchRef.current) {
+      searchRef.current.value = option.label;
+    }
+  };
+
+  useEffect(() => {
+    if (dropdownRef.current) {
```
Comment (hachiojidev) [inline packages/pancake-uikit/src/components/DropdownSearch/DropdownSearch.tsx:130]: Please put `dropdownRef` in the dependency list

File: packages/pancake-uikit/src/components/DropdownSearch/types.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export interface DropdownSearchProps {
+  options: DropdownSearchOptionProps[];
+  onChange?: (option: DropdownSearchOptionProps) => void;
+}
+
+export interface DropdownSearchOptionProps {
+  label: string;
+  value: any;
```
Comment (hachiojidev) [inline packages/pancake-uikit/src/components/DropdownSearch/types.ts:8]: Number or string? If so `ReactText`.

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: packages/pancake-uikit/src/components/DropdownSearch/DropdownSearch.tsx (hunk 1 above)
Comment (memoyil) [reply packages/pancake-uikit/src/components/DropdownSearch/DropdownSearch.tsx]: Removed

File: packages/pancake-uikit/src/components/DropdownSearch/DropdownSearch.tsx (hunk 2 above)
Comment (memoyil) [reply packages/pancake-uikit/src/components/DropdownSearch/DropdownSearch.tsx:129]: Added

File: packages/pancake-uikit/src/components/DropdownSearch/types.ts (hunk 3 above)
Comment (memoyil) [reply packages/pancake-uikit/src/components/DropdownSearch/types.ts]: Done

**Comment by memoyil** [comment]
@hachiojidev you can review it again

**Pull request closed without merging**